packages/server/tests/acceptance/htsPrecompile/tokenManagement.spec.ts

```diff
@@ -327,40 +324,40 @@ describe('@tokenmanagement HTS Precompile Token Management Acceptance Tests', as
+      const tokenInfoBefore = ((await txBeforeInfo.wait()).logs.filter(e => e.fragment.name === Constants.HTS_CONTRACT_EVENTS.TokenInfo)[0].args.tokenInfo)[0];
 
-      //updating only token info, not token keys
-      const token = {
-        ...tokenInfoBefore, tokenKeys: []
-      };
+      // updating only token info, not token keys
+      const token = Utils.convertEthersResultIntoStringsArray(tokenInfoBefore);
+      token[7] = [];
 
       setUpdatedValues(token);
 
       // update contract properties
       const txUpdate = await mainContractOwner.updateTokenInfoPublic(HTSTokenContractAddress, token, Constants.GAS.LIMIT_1_000_000);
-      expect((await txUpdate.wait()).events.filter(e => e.event === Constants.HTS_CONTRACT_EVENTS.ResponseCode)[0].args.responseCode).to.be.equal(TX_SUCCESS_CODE);
+      expect((await txUpdate.wait()).logs.filter(e => e.fragment.name === Constants.HTS_CONTRACT_EVENTS.ResponseCode)[0].args.responseCode).to.be.equal(TX_SUCCESS_CODE);
 
       const txAfterInfo = await mainContract.getTokenInfoPublic(HTSTokenContractAddress, Constants.GAS.LIMIT_1_000_000);
-      const tokenInfoAfter = ((await txAfterInfo.wait()).events.filter(e => e.event === Constants.HTS_CONTRACT_EVENTS.TokenInfo)[0].args.tokenInfo)[0];
+      const tokenInfoAfter = ((await txAfterInfo.wait()).logs.filter(e => e.fragment.name === Constants.HTS_CONTRACT_EVENTS.TokenInfo)[0].args.tokenInfo)[0];
       await checkUpdatedTokenInfo(tokenInfoAfter);
     });
 
     it('should update non-fungible token properties', async function () {
       const txBeforeInfo = await mainContract.getTokenInfoPublic(NftHTSTokenContractAddress, Constants.GAS.LIMIT_1_000_000);
-      const tokenInfoBefore = ((await txBeforeInfo.wait()).events.filter(e => e.event === Constants.HTS_CONTRACT_EVENTS.TokenInfo)[0].args.tokenInfo)[0];
+      const tokenInfoBefore = ((await txBeforeInfo.wait()).logs.filter(e => e.fragment.name === Constants.HTS_CONTRACT_EVENTS.TokenInfo)[0].args.tokenInfo)[0];
 
-      //updating only token info, not token keys
-      const token = {
-        ...tokenInfoBefore, tokenKeys: []
-      };
+      // updating only token info, not token keys
+      const token = Utils.convertEthersResultIntoStringsArray(tokenInfoBefore);
+      token[7] = [];
 
       setUpdatedValues(token);
 
       const txUpdate = await mainContractOwner.updateTokenInfoPublic(NftHTSTokenContractAddress, token, Constants.GAS.LIMIT_1_000_000);
-      expect((await txUpdate.wait()).events.filter(e => e.event === Constants.HTS_CONTRACT_EVENTS.ResponseCode)[0].args.responseCode).to.be.equal(TX_SUCCESS_CODE);
+      expect((await txUpdate.wait()).logs.filter(e => e.fragment.name === Constants.HTS_CONTRACT_EVENTS.ResponseCode)[0].args.responseCode).to.be.equal(TX_SUCCESS_CODE);
+
+      await new Promise(r => setTimeout(r, 2000));
```

Comment:
Q: why do we need a timeout?
Should the wait on the transaction and it's receipt be enough?

Reply:
The receipt should be enough, will remove it, I guess it's a leftover from testing.